src/main/java/carpet/script/value/EntityValue.java

```diff
@@ -891,21 +893,51 @@ else if (v instanceof ListValue)
             throw new InternalExpressionException("'effect' needs either no arguments (clear) or effect name, duration, and optional amplifier, show particles and show icon");
         });
 
-        // gamemode
-        // spectate
-        // "fire"
-        // "extinguish"
-        // "silent"
-        // "gravity"
-        // "invulnerable"
-        // "dimension"
-        // "item"
-        // "count",
-        // "age",
-        // "effect_"name
-        // "hold"
-        // "hold_offhand"
-        // "jump"
+        put("gamemode_id", (e,v)->{
+            if(!(e instanceof ServerPlayerEntity)){
+                throw new InternalExpressionException("Cannot change the gamemode of non-player entities");
+            }
+            if(v.readInteger()==0){
+                ((ServerPlayerEntity) e).setGameMode(GameMode.SURVIVAL);
+                return;
+            }
+            if(v.readInteger()==1){
+                ((ServerPlayerEntity) e).setGameMode(GameMode.CREATIVE);
+                return;
+            }
+            if(v.readInteger()==3){
+                ((ServerPlayerEntity) e).setGameMode(GameMode.SPECTATOR);
+                return;
+            }
+            if(v.readInteger()==2){
+                ((ServerPlayerEntity) e).setGameMode(GameMode.ADVENTURE);
+                return;
+            }
+            throw new InternalExpressionException("s"+v.getString()+"s is an invalid gamemode type");
+        });
+
+        put("jump",(e,v)->{
```

Comment:
all living entities can jump

Reply:
it wont let me unless I typecast to PlayerEntity. Maybe I'm using the wrong jump method but It just wont work

Reply:
when I typecast to LivingEntity, for which there is a jump method, it tells me `The method jump() from the type LivingEntity is not visible`

Reply:
`protected void jump() {
  }`

Reply:
in LivingEntity.class

Reply:
I dont see how to do it without making EntityValue extend LivingEntity, which cant happen

Reply:
DOne, but no it needs to be changed to 'jumping'